sarsi-elm/Main.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE Rank2Types #-}
+module Main where
+
+import Codec.ELM.Log (decodeMessage, toHandle)
+import Codec.Sarsi (Event(..), Level(..), Message(..))
+import Codec.Sarsi.ELM (fromELMLog)
+import Data.Machine (ProcessT, (<~), asParts, auto, autoM, prepended, runT_, source)
+import Data.List (foldl')
+import Sarsi (getBroker, getTopic)
+import Sarsi.Producer (produce)
+import System.Environment (getArgs)
+import System.IO (stderr)
+import System.IO.Machine (byLine)
+import System.Process (StdStream(..), shell, std_in, std_out)
+import System.Process.Machine (callProcessMachines, mStdOut)
+import System.Exit (ExitCode, exitWith)
+
+import qualified Data.List as List
+import qualified Data.Text as Text
+import qualified Sarsi as Sarsi
+
+title :: String
+title = concat [Sarsi.title, "-elm"]
+
+-- TODO Use Wee/Tee to merge stdout/stderr (keep echoText splitted).
```

Comment:
I'm currently implementing `sarsi-rs`, and I'll need the same code, I'll try to extract it in the library.